### PR TITLE
feat(md-whiteframe): Create Whiteframe

### PR DESCRIPTION
**Pull request by bakgat**
The MdWhiteframe component based on angular 1.x version.

### CODE REVIEW

File: src/components/whiteframe/whiteframe.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    ngOnInit() {
+        this.elevation_ = parseInt(this.mdWhiteframe, 10) || MdWhiteframe.DEFAULT_DP;
+
+        if (this.elevation_ > MdWhiteframe.MAX_DP || this.elevation_ < MdWhiteframe.MIN_DP) {
+            this.elevation_ = MdWhiteframe.DEFAULT_DP;
+        }
+
+        this.dpCssClass = 'md-whiteframe-' + this.elevation_ + 'dp';
+        this.el.nativeElement.className +=
+            (this.el.nativeElement.className !== '' ? ' ' : '') +
```
Comment (DevVersion) [inline src/components/whiteframe/whiteframe.ts]: Why you append the class to `className`?. `classList` will do the same in a native element. ([here](https://developer.mozilla.org/de/docs/Web/API/Element/classList)

Better use:
```
this.el.nativeElement.className.add(this.dpCssClass)
```

File: src/components/whiteframe/whiteframe.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import {Component, Input, ElementRef,ViewEncapsulation, OnInit} from 'angular2/core';
```
Comment (DevVersion) [inline src/components/whiteframe/whiteframe.ts]: Little style thing - You should space the `ViewEncpsulation` from the comma.

File: src/components/whiteframe/whiteframe.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@Component({
+    selector: '[md-whiteframe]',
+    templateUrl: './components/whiteframe/whiteframe.html',
+    styleUrls: ['./components/whiteframe/whiteframe.css'],
+    encapsulation: ViewEncapsulation.None
+})
+export class MdWhiteframe implements OnInit {
+    @CONST() static MIN_DP:number = 1;
+    @CONST() static MAX_DP:number = 24;
+    @CONST() static DEFAULT_DP:number = 4;
```
Comment (DevVersion) [inline src/components/whiteframe/whiteframe.ts]: I also suggest spacing the type definition from the constant (to have the same code-style as in other components). 
Like so: `@CONST() static DEFAULT_DP: number = 4;`

File: src/components/whiteframe/whiteframe.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<ng-content></ng-content>
```
Comment (DevVersion) [inline src/components/whiteframe/whiteframe.html]: I'm not sure about using a component for the `whiteframe`. Wrapping the content in a component template makes not really a sense. Thoughts?

File: src/components/whiteframe/whiteframe.scss [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+.md-whiteframe-7dp, .md-whiteframe-z3 {
+  box-shadow: $whiteframe-shadow-7dp;
+}
+.md-whiteframe-8dp {
+  box-shadow: $whiteframe-shadow-8dp;
+}
+.md-whiteframe-9dp {
+  box-shadow: $whiteframe-shadow-9dp;
+}
+.md-whiteframe-10dp, .md-whiteframe-z4 {
```
Comment (DevVersion) [inline src/components/whiteframe/whiteframe.scss]: I would not start introducing the `zN` whiteframe classes. These are deprecated and will be soon removed from `material1`

File: src/components/whiteframe/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# MdWhiteframe
+
+MdWhiteframe is a component that applies an elevation shadow to an element.
+
+## Screenshot
+<img src="http://www.klimtoren.be/content/material2/mdwhiteframe-example.png">
```
Comment (DevVersion) [inline src/components/whiteframe/README.md]: Also hosting the screenshot on your private server is not a good idea I think.

File: src/components/whiteframe/whiteframe.ts (hunk 1 above)
Comment (bakgat) [inline src/components/whiteframe/whiteframe.ts]: Done. Changed to this.el.nativeElement.classList.add(this.dpCssClass)

File: src/components/whiteframe/whiteframe.ts (hunk 2 above)
Comment (bakgat) [inline src/components/whiteframe/whiteframe.ts]: Thank you, done.

File: src/components/whiteframe/whiteframe.ts (hunk 3 above)
Comment (bakgat) [inline src/components/whiteframe/whiteframe.ts]: How about `private elevation_:number;` ?

(I'm glad to get all the feedback. I've never contributed any project, so it's very new for me :smile: )

File: src/components/whiteframe/whiteframe.scss (hunk 5 above)
Comment (bakgat) [inline src/components/whiteframe/whiteframe.scss]: Removed zN classes.

File: src/components/whiteframe/README.md (hunk 6 above)
Comment (bakgat) [inline src/components/whiteframe/README.md]: Where do I host screenhost better?

File: src/components/whiteframe/whiteframe.html (hunk 4 above)
Comment (bakgat) [inline src/components/whiteframe/whiteframe.html]: I believe Directive whould be better. But where should the css classes be loaded then?

File: src/components/whiteframe/README.md (hunk 6 above)
Comment (DevVersion) [inline src/components/whiteframe/README.md]: The Material team itself host their images [there](https://github.com/angular/code.material.angularjs.org). So I think they will host the screenshot their self.

File: src/components/whiteframe/whiteframe.html (hunk 4 above)
Comment (DevVersion) [inline src/components/whiteframe/whiteframe.html]: Yes that will be a problem. We need a feedback from one of the team.

File: src/components/whiteframe/README.md (hunk 6 above)
Comment (bakgat) [inline src/components/whiteframe/README.md]: Do I remove the screenshot then?

Comment (DevVersion) [inline src/components/whiteframe/README.md]: No keep it temporary

**Comment by jelbourn** [comment]
@bakgat Please read our [contributing guidelines](https://github.com/angular/material2/blob/master/CONTRIBUTING.md).

Before beginning work on any new features, please open an issue so we can discuss before anyone starts writing code. At this point, we're not sure `whiteframe` is one of the components we want to include in material2.

**Comment by bakgat** [comment]
@jelbourn Well, I've learned a lot about contributing today. :smile: 
Is any contributing accepted right now? Because I've coded some basic components already to get my own apps up and running.

**Comment by jelbourn** [comment]
@bakgat  Yes, we'd love to have contributions! It's important to go through the documented process, though, so that everyone is on the same page. The first step is opening an issue with the feature you're proposing and we can go from there.

**Pull request closed without merging**